modules/build/src/main/scala/scala/build/internal/JavaParserProxy.scala

```diff
@@ -0,0 +1,65 @@
+package scala.build.internal
+
+import coursier.cache.ArchiveCache
+import coursier.util.Task
+
+import scala.build.EitherCps.{either, value}
+import scala.build.Logger
+import scala.build.errors.BuildException
+import scala.util.Properties
+
+/** Downloads and runs java-class-name as an external binary.
+  *
+  * See [[JavaParserProxyJvm]] for the override that runs things in memory using java-class-name
+  * from the class path.
+  */
+class JavaParserProxy {
+
+  /** Extracts the class name of a Java source, using the dotty Java parser.
+    *
+    * @param archiveCache
+    *   when running java-class-name as a binary (see [[JavaParserProxyJvm]]), this cache is used to
+    *   download the binary
+    * @param content
+    *   the Java source to extract a class name from
+    * @param logger
+    *   a logger to log messages
+    * @return
+    *   either some class name (if one was found) or none (if none was found), or a
+    *   [[BuildException]]
+    */
+  def className(
+    archiveCache: ArchiveCache[Task],
+    content: Array[Byte],
+    logger: Logger
+  ): Either[BuildException, Option[String]] = either {
+
+    val versionOpt = Some("0.1.0")
```

Comment:
wouldn't this always be picked over `Constants.javaClassNameVersion`, given how it's coded atm?
Why not just do `val version = Constants.javaClassNameVersion`?

Reply:
Should be fixed.